fstat/parser.py

```diff
@@ -6,31 +6,44 @@
 from fstat import app, db, Failure, FailureInstance
 
 
+def save_failure(signature, url, job_name, build_info):
+    failure = Failure.query.filter_by(
+            signature=signature).first()
+    # If it doesn't exist, create a job first
+    if failure is None:
+        failure = Failure(signature=signature)
+    failure_instance = FailureInstance(url=url,
+                                       job_name=job_name)
+    failure_instance.process_build_info(build_info)
+    failure_instance.failure = failure
+    try:
+        db.session.add(failure)
+        db.session.add(failure_instance)
+        db.session.commit()
+    except IntegrityError:
+        db.session.rollback()
+
+
 def process_failure(url, job_name, build_info):
     text = requests.get(url, verify=False).text
     accum = []
+    if text.find("Finished: ABORTED") != -1:
```

Comment:
Also if possible can you give me the example of job that was aborted due to new patchset was uploaded.